src/plugins_types.c

```diff
@@ -1070,8 +1070,21 @@ lyplg_type_resolve_leafref(const struct lysc_type_leafref *lref, const struct ly
         }
         goto cleanup;
     }
-    if (target) {
-        *target = set.val.nodes[i].node;
+    if (targets) {
+        LY_CHECK_GOTO(ly_set_new(targets), cleanup);
```

Comment:
Should be `LY_CHECK_GOTO(rc = ly_set_new(targets), cleanup);` to return an error.

Reply:
Done